### PR TITLE
numa_topology_with_hugepage: add case numa topology with hugepage

### PR DESCRIPTION
**Pull request by dzhengfy**
Scenario: Test vm numa backend with different hugepage size

Case ID:  VIRT-297086

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_numa_topology.numa_topology_with_hugepage
JOB ID     : c277928de6d3ef5b620a66f7016a4041c60f8725
JOB LOG    : /var/lib/avocado/job-results/job-2023-06-11T08.23-c277928/job.log
 (1/4) type_specific.io-github-autotest-libvirt.guest_numa_topology.numa_topology_with_hugepage.2M: PASS (62.43 s)
 (2/4) type_specific.io-github-autotest-libvirt.guest_numa_topology.numa_topology_with_hugepage.1G: PASS (61.87 s)
 (3/4) type_specific.io-github-autotest-libvirt.guest_numa_topology.numa_topology_with_hugepage.2M_1G: PASS (66.10 s)
 (4/4) type_specific.io-github-autotest-libvirt.guest_numa_topology.numa_topology_with_hugepage.scarce_mem: PASS (24.92 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-06-11T08.23-c277928/results.html
JOB TIME   : 217.60 s
```

Tested with libvirt-9.4.0

**Comment by dzhengfy** [comment]
Depend on 
- https://github.com/autotest/tp-libvirt/pull/4912

**Comment by dzhengfy** [comment]
Depend on 
- https://github.com/avocado-framework/avocado-vt/pull/3698

File: libvirt/tests/src/numa/guest_numa_topology/numa_topology_with_hugepage.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+def setup_default(test_obj):
+    """
+    Default setup function for the test
+
+    :param test_obj: NumaTest object
+    """
+    test_obj.setup()
+    params_2M = {'hugepage_size': '2048',
+                 'hugepage_path': test_obj.params.get('hugepage_path_2M'),
```
Comment (nanli1) [inline libvirt/tests/src/numa/guest_numa_topology/numa_topology_with_hugepage.py:39]: Here depends on https://github.com/avocado-framework/avocado-vt/pull/3698/files , Could you please add depend here

**Review by nanli1 — APPROVED** [review]

File: libvirt/tests/src/numa/guest_numa_topology/numa_topology_with_hugepage.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+from virttest import virsh
+from virttest import test_setup
+from virttest.libvirt_xml import vm_xml
+from virttest.utils_test import libvirt
+
+from provider.numa import numa_base
+
+
+def setup_hugepage(params):
+
```
Comment (Yingshun) [inline libvirt/tests/src/numa/guest_numa_topology/numa_topology_with_hugepage.py:25]: plz add docstring, thx!

**Review by Yingshun — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/numa/guest_numa_topology/numa_topology_with_hugepage.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/numa/guest_numa_topology/numa_topology_with_hugepage.py]: Updated, Thanks

**Review by Yingshun — APPROVED** [review]

**Pull request merged**